app/code/Magento/Catalog/Model/Product/Gallery/Processor.php

```diff
@@ -183,6 +184,13 @@ public function addImage(
         $attrCode = $this->getAttribute()->getAttributeCode();
         $mediaGalleryData = $product->getData($attrCode);
         $position = 0;
+
+        $absoluteFilePath = $this->mediaDirectory->getAbsolutePath($file);
+        $imageMimeType = mime_content_type($absoluteFilePath);
+        $imageContent = file_get_contents($absoluteFilePath);
+        $imageBase64 = base64_encode($imageContent);
+        $imageName = pathinfo($destinationFile, PATHINFO_FILENAME);
```

Comment:
PHP native functions must not be used outside of Magento Framework. Use existing framework components or introduce new ones when necessary.

Reply:
Fair enough, thanks for note. I removed all except base64 & a pathinfo because we already have a lot of usages of that.